CONTRIBUTING.md

```diff
@@ -0,0 +1,85 @@
+# Contributing to Cyrus
+
+So, you want to contribute to Cyrus?  Great!
+
+You'll probably want to join the [cyrus-devel mailing
+list](https://www.cyrusimap.org/imap/support/feedback-mailing-lists.html#feedback-mailing-lists)
+where development issues get discussed.  You don't need to, but if you're
+considering doing a substantial amount of work, it's a good idea to ask about
+it first.
+
+This document is meant to be a quick overview of the most important things you
+need to know to get your work reviewed, approved, and into Cyrus.
+
+## Your Code
+
+Cyrus doesn't have a hard and fast style guide, *but it will*.  For now,
+consult the [Cyrus hacking
+docs](https://www.cyrusimap.org/imap/developer/guidance/hacking.html), which
+spell out some of the standards of formatting and construction.
```

Comment:
I wonder if we should say something about "copy the style of the surrounding code"?  I feel like this should go without saying, and yet.

I feel like this PR should also add a note: or warning: box to the top of hacking.rst explaining that it's very out of date, which might be an alternate place to say "copy the style of the surrounding code".

Reply:
I have added that to this PR in both the guide and in this document.